### PR TITLE
Support EIP-1271 signatures

### PR DESCRIPTION
**Pull request by mdehoog**
Ballot submission returns an error for smart contract wallets, because the regex enforces the signature to be 130 hex characters (65 bytes), but EIP-1271 signatures are often far longer than that.

This PR fixes the signature validation bug by removing the exact signature length validation.

~Unfortunately the `ballotSubmittions.signature` column in the schema is a VARCHAR(255) which limits the ability to store full signatures in the database. Not sure what the downstream effects of this are, but I've updated the upsert to truncate the signature to fit in the column. If that breaks something downstream, we'll have to update the schema itself (i think https://github.com/voteagora/queries is a private repo?).~

### CODE REVIEW

**Comment by mdehoog** [comment]
@stepandel @yitongzhang any chance we can merge and deploy this before `Oct 14th 18:00 UTC`? I'm currently blocked on voting for round5, would love to get it in prior to the round close. Thanks.

**Comment by yitongzhang** [comment]
Hey @mdehoog good to see you again. It's been a bit! Sorry we totally missed this PR last Friday. It was a hectic day. For time-sensitive issues, please ping us or the OP foundation directly on slack or via twitter DM's.

Just flagged this PR to our eng team to make sure they're aware and can address this issue.

**Review by stepandel — CHANGES_REQUESTED** [review]
please remove signature tranc and leave the zod validator changes.
Thank you!

File: src/app/api/common/ballots/submitBallot.ts [hunk 1]
```diff
@@ -38,6 +38,13 @@ async function submitBallotForAddress({
     throw new Error("Invalid signature");
   }
 
+  // The signature column in the schema is VARCHAR(255). EIP-1271 signatures
+  // can be longer than that, so we truncate them to fit in the database.
+  let signature = data.signature;
+  if (signature.length > 255) {
+    signature = signature.substring(0, 252) + "...";
+  }
+
```
[the remark this answers is not in the document]
Comment (mdehoog) [reply src/app/api/common/ballots/submitBallot.ts]: done, thanks!

**Review by stepandel — APPROVED** [review]
beauty, thank you!

**Comment by kent** [comment]
Thank you @mdehoog, we are upping our GitHub repo notifications to make sure we don't sleep on these. Awesome to have you as our first non Agora contributor :) tysm

**Pull request merged**